cmd/prowjob-controller/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	controller "github.com/openshift/release-controller/pkg/controller/prowjob"
+	"github.com/openshift/release-controller/pkg/signals"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/klog/v2"
+	clientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	informers "k8s.io/test-infra/prow/client/informers/externalversions"
+	"time"
+)
+
+var (
+	clusterContext string
+	authInfo       string
+	prowNamespace  string
+)
+
+func main() {
+	var cfg *rest.Config
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	// set up signals, so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler(cancelFunc)
+
+	overrides := clientcmd.ConfigOverrides{Context: api.Context{Cluster: clusterContext, AuthInfo: authInfo}}
+	cfg, err := loadClusterConfig(&overrides)
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building kubernetes clientset: %s", err.Error())
+	}
+
+	prowJobClientSet, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building prowjob clientset: %s", err.Error())
+	}
+
+	prowJobInformerFactory := informers.NewSharedInformerFactory(prowJobClientSet, 5*time.Minute)
+	prowJobController := controller.NewProwJobController(ctx, cancelFunc, kubeClient, prowJobInformerFactory.Prow().V1().ProwJobs(), prowNamespace)
+
+	prowJobInformerFactory.Start(stopCh)
+
+	if err = prowJobController.Run(stopCh); err != nil {
+		klog.Fatalf("Error running prowjob controller: %s", err.Error())
+	}
+}
+
+func init() {
+	original := flag.CommandLine
+	original.Set("alsologtostderr", "true")
+	original.Set("v", "2")
+	klog.InitFlags(original)
+	flag.StringVar(&clusterContext, "cluster-context", "", "The cluster context to use (defaults to the current context if not specified).")
+	flag.StringVar(&authInfo, "auth-info", "", "The cluster auth info to use (defaults to the current context if not specified).")
+	flag.StringVar(&prowNamespace, "prow-namespace", "ci", "The namespace where the Prow jobs will be created.")
+	flag.Parse()
+}
+
+func loadClusterConfig(overrides *clientcmd.ConfigOverrides) (*rest.Config, error) {
+	cfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), overrides)
+	clusterConfig, err := cfg.ClientConfig()
+	if err != nil {
```

Comment:
nit: you can just `return cfg.ClientConfig()` here, and if you do that, this `loadClusterConfig` method is obsolete?

Reply:
Yes, and yes.  I've removed the extra logic.